### PR TITLE
new docker image to open jupyterlab and restrict download

### PR DESCRIPTION
**Pull request by urvi-occ**
Jira Ticket: [OCC-69](https://ctds-planx.atlassian.net/browse/OCC-69)

### New Features
Added a new docker container image which restricts the download of any notebook/data on jupyter lab (docker image is referenced on [jupyterslim](https://github.com/uc-cdis/jupyter-superslim/blob/master/jupyter-superslim/Dockerfile)

### Improvements
Updated Python version to 3.9 and updated version for conda_version and miniforge_patch_number

[OCC-69]: https://ctds-planx.atlassian.net/browse/OCC-69?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: jupyter-covid19/Dockerfile [hunk 1]
```diff
@@ -96,5 +96,6 @@ ADD --chown=jovyan:users https://raw.githubusercontent.com/uc-cdis/covid19-tools
 RUN touch /home/$NB_USER/covid19-notebook/peregrine.py
 
 # premade notebooks dependencies
-RUN pip install --upgrade 'pip<20.3' # pip 20.3 causes dependency resolution issues
-RUN pip install -r /home/$NB_USER/covid19-notebook/requirements.txt
+# RUN pip install --upgrade 'pip<20.3' # pip 20.3 causes dependency resolution issues
+# COPY requirements.txt /home/$NB_USER/covid19-notebook/requirements.txt
+# RUN pip install -r /home/$NB_USER/covid19-notebook/requirements.txt --upgrade
```
Comment (paulineribeyre) [inline jupyter-covid19/Dockerfile:101]: why did you remove this?

File: jupyter-covid19-restrict-download/start-notebook.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+set -e
```
Comment (paulineribeyre) [inline jupyter-covid19-restrict-download/start-notebook.sh:5]: @urvi-occ where did you copy `start-notebook.sh`, `start-singleuser.sh` and `start.sh` from and why do we need them?

File: jupyter-covid19-restrict-download/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,90 @@
+ARG ROOT_CONTAINER=quay.io/cdis/ubuntu:focal
```
Comment (paulineribeyre) [inline jupyter-covid19-restrict-download/Dockerfile:1]: - can you rename `jupyter-covid19-restrict-download` to `jupyter-restricted-download` as mentioned in the ticket comments (since it's not specific to PRC but just a new base image),
- and add [this workflow](https://github.com/uc-cdis/containers/blob/copy-image-jupyter-covid19-restrict-download/.github/workflows/build_jupyter_covid19_restricted_download.yml) that i had shared in the ticket as well (the file paths will need to be renamed there too)?

File: jupyter-covid19-restrict-download/jupyterslim.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+version: '3'
```
Comment (paulineribeyre) [inline jupyter-covid19-restrict-download/jupyterslim.yaml:1]: let's delete this file

File: jupyter-covid19-restrict-download/README.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ipykernel        : 5.3.0
+jupyter client   : 6.1.3
+jupyter lab      : 2.1.3
+nbconvert        : 5.6.1
+ipywidgets       : 7.5.1
+nbformat         : 5.0.6
+traitlets        : 4.3.3
+```
+
+Note: the versions shown above may not apply to current image versions.
```
Comment (paulineribeyre) [inline jupyter-covid19-restrict-download/README.md:52]:
```suggestion
# TL;DR

A "restricted" build of `jupyter-slim`.  Basically the same as `../jupyter-slim/`, but it restricts user to download anything from the jupyter lab
```

File: jupyter-covid19-restrict-download/start-notebook.sh (hunk 2 above)
Comment (urvi-occ) [reply jupyter-covid19-restrict-download/start-notebook.sh:5]: As I referenced in the PR description, I am referencing the whole image on [jupyterslim](https://github.com/uc-cdis/jupyter-superslim/blob/master/jupyter-superslim/Dockerfile) as I had to update python version

File: jupyter-covid19/Dockerfile (hunk 1 above)
Comment (urvi-occ) [reply jupyter-covid19/Dockerfile:101]: This is a mistake, let me uncomment this and commit it again. I would still have the line `# RUN pip install --upgrade 'pip<20.3' # pip 20.3 causes dependency resolution issues` commented as it is having issues with new version of python for me

File: jupyter-covid19-restrict-download/Dockerfile (hunk 3 above)
Comment (urvi-occ) [reply jupyter-covid19-restrict-download/Dockerfile:1]: Okay. You should be able to see the changes in the upcoming commit

File: jupyter-covid19-restrict-download/README.md (hunk 5 above)
Comment (paulineribeyre) [reply jupyter-covid19-restrict-download/README.md:52]: Hi @urvi-occ, I think maybe you forgot to push changes?

File: jupyter-covid19-restrict-download/Dockerfile (hunk 3 above)
Comment (urvi-occ) [reply jupyter-covid19-restrict-download/Dockerfile]: I have changed the workflow file but the file name is still build_jupyter_covid19_restricted_download.yml
as I didn't know if I could change the file or not

**Comment by paulineribeyre** [comment]
This was done in https://github.com/uc-cdis/containers/pull/145, https://github.com/uc-cdis/containers/pull/146, https://github.com/uc-cdis/containers/commit/f541afb3e28a3bbd706039223c5e1fd943956014 and https://github.com/uc-cdis/containers/commit/78b8c2b90e36f7edcc32bbf62e245b452ac88a33

Deployed with https://github.com/uc-cdis/gitops-qa/commit/6d54d3f40aae2158dbe8a18215a7e09b152a7d57
